rustfmt-core/rustfmt-lib/src/visitor.rs

```diff
@@ -364,7 +364,7 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
                     mk_sp(s.lo(), b.span.lo()),
                 )
             }
-            _ => unreachable!(),
+            _ => return,
```

Comment:
I think we probably still want `unreachable!` on the [Closure variant](https://docs.rs/rustc-ap-syntax/642.0.0/rustc_ap_syntax/visit/enum.FnKind.html#variant.Closure) here, but just not when the Fn block is `None`?

What about this:
```suggestion
            visit::FnKind::Fn(_, _, _, _, None) => return,
            _ => unreachable!(),
```

Reply:
Thank you for the review. I fixed it.